linux_os/guide/services/ssh/ssh_server/sshd_use_priv_separation/oval/shared.xml

```diff
@@ -1,36 +1 @@
-<def-group>
-  <definition class="compliance" id="sshd_use_priv_separation" version="1">
-    <metadata>
-      <title>Rule title of sshd_use_priv_separation</title>
-      <affected family="unix">
-        <platform>multi_platform_all</platform>
-      </affected>
-      <description>Ensure 'UsePrivilegeSeparation' is configured with value 'sandbox' in '/etc/ssh/sshd_config'</description>
-    </metadata>
-    <criteria comment="sshd is configured correctly or is not installed"
-    operator="OR">
-      <criteria comment="sshd is not installed" operator="AND">
-        <extend_definition comment="sshd is not required or requirement is unset"
-        definition_ref="sshd_not_required_or_unset" />
-        {{% if product in ['opensuse', 'sle11', 'sle12'] %}}
-        <extend_definition comment="rpm package openssh removed"
-        definition_ref="package_openssh_removed" />
-        {{% else %}}
-        <extend_definition comment="rpm package openssh-server removed"
-        definition_ref="package_openssh-server_removed" />
-        {{% endif %}}
-      </criteria>
-      <criterion comment="Check the UsePrivilegeSeparation in /etc/ssh/sshd_config"
-      test_ref="test_sshd_use_priv_separation" />
-    </criteria>
-  </definition>
-  
-  {{{ oval_line_in_file_test(path='/etc/ssh/sshd_config', parameter='UsePrivilegeSeparation') }}}
-  {{{ oval_line_in_file_object(path='/etc/ssh/sshd_config', prefix_regex="^[ \\t]*(?i)", parameter='UsePrivilegeSeparation', separator_regex='(?-i)[ \\t]+') }}}
-  
-  <ind:textfilecontent54_state id="state_sshd_use_priv_separation" version="1">
-    <ind:subexpression operation="equals" var_ref="var_sshd_priv_separation" />
-  </ind:textfilecontent54_state>
-
-  <external_variable comment="external variable for the desired value of UsePrivilegeSeparation" datatype="string" id="var_sshd_priv_separation" version="1" />
-</def-group>
+{{{ oval_sshd_config(parameter="UsePrivilegeSeparation", value="(sandbox|yes)", missing_parameter_pass=true) }}}
```

Comment:
This isn't going to work. If `var_sshd_priv_separation` is set to `sandbox` and `sshd_config` isn't configured, pretty sure the oval will pass which means that the pass is really a false positive when `sandbox` is used.

Reply:
From the STIG, both `sandbox` and `yes` values should be accepted and `yes` is the default value of `UsePrivilegeSeparation`, so that's why `missing_parameter_pass=true`

[V-72265](
https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_7_security_technical_implementation_guide/V-72265?version=v2r7) 
`Uncomment the "UsePrivilegeSeparation" keyword in "/etc/ssh/sshd_config" (this file may be named differently or be in a different location if using a version of SSH that is provided by a third-party vendor) and set the value to "sandbox" or "yes":`

[sshd_config](https://linux.die.net/man/5/sshd_config)
`UsePrivilegeSeparation`
`Specifies whether sshd(8) separates privileges by creating an unprivileged child process to deal with incoming network traffic. After successful authentication, another process will be created that has the privilege of the authenticated user. The goal of privilege separation is to prevent privilege escalation by containing any corruption within the unprivileged processes. The default is ''yes''. `

Reply:
One thing I noticed is that if this change is correct, we need to update the rule text to explicitly mention that both options are accepted.

Reply:
Doesn't matter. If I purposefully choose `sandbox` which affords more protections around child process sandboxing, and `UsePrivilegeSeparation` is not configured, it is a false positive because it's default is `yes` and less secure which means my scan and scan results are wrong.

Reply:
Ok agreed, so I'll keep both check and remediation to expect for the variable `var_sshd_priv_separation`.